### DISCUSSION TITLE
ViT vs ResNet for object recognition

### DISCUSSION THREAD

**Discussion by adambeck7** [immich-app/immich#3921] [Q&A] [answered]
I noticed the CLIP model is using ViT, but the Image Detection is using ResNet. Would I stand to gain anything by moving to a ViT model like `google/vit-base-patch16-224`? My server has plenty of power, so I'm more concerned about accuracy than performance when tagging.

**Comment by mertalev** [comment, marked as answer]
ViT models are much more accurate than Resnet50, so the tags you get will be better. The only reason I'm keeping Resnet50 as default is to not increase the default memory  usage.

**Reply by bo0tzz** [reply]
Do you have a ballpark for the RAM difference between the two?

**Reply by adambeck7** [reply]
@bo0tzz mine looks like it has topped out at 2.55gb of ram when rerunning the object detection. Before starting the job, the container was 630mb, but not sure how much that matters if the job wasn't running. Provided I didn't need to restart the container between switching models.  Let me know if I should have. The switch seemed a little too quick and easy :)

**Reply by mertalev** [reply]
You don't need to restart since the new model name is sent with the request. But if you ran the "Tag Objects" job before changing the model, the old model will still be loaded in the ML container until you restart.

**Reply by adambeck7** [reply]
Good to know, thanks. Incredible work on this stuff. 

I switched the model, saved, waited ~5 min then ran the object detection.

**Reply by mertalev** [reply]
No need to wait either! It applies right away.

> Do you have a ballpark for the RAM difference between the two?

`google/vit-base-patch16-224` uses about 100mb more, which is actually a smaller difference than I was expecting. Might be worth considering the switch.
